net/process.go

```diff
@@ -29,57 +30,96 @@ import (
 	"github.com/sourcenetwork/defradb/errors"
 	"github.com/sourcenetwork/defradb/events"
 	"github.com/sourcenetwork/defradb/logging"
-	"github.com/sourcenetwork/defradb/merkle/clock"
 	"github.com/sourcenetwork/defradb/merkle/crdt"
 )
 
+type blockProcessor struct {
+	*Peer
+	// List of composite blocks to eventually merge
+	composites *list.List
+}
+
+func newBlockProcessor(p *Peer) *blockProcessor {
+	return &blockProcessor{
+		Peer:       p,
+		composites: list.New(),
+	}
+}
+
 // processNode is a general utility for processing various kinds
 // of CRDT blocks
-func (p *Peer) processLog(
+func (bp *blockProcessor) processRemoteBlock(
 	ctx context.Context,
+	session *sync.WaitGroup,
 	txn datastore.Txn,
-	col client.Collection,
 	dsKey core.DataStoreKey,
-	field string,
 	nd ipld.Node,
 	getter ipld.NodeGetter,
-	removeChildren bool,
-) ([]cid.Cid, error) {
+	isComposite bool,
+) error {
 	log.Debug(ctx, "Running processLog")
 
-	crdt, err := initCRDTForType(ctx, txn, col, dsKey, field)
-	if err != nil {
-		return nil, err
+	if err := txn.DAGstore().Put(ctx, nd); err != nil {
+		return err
 	}
 
-	delta, err := crdt.DeltaDecode(nd)
-	if err != nil {
-		return nil, errors.Wrap("failed to decode delta object", err)
+	if isComposite {
+		bp.composites.PushFront(nd)
```

Comment:
todo:  This feels quite strange, as you are appending to a private property on an object that never reads from it (the only reading of it is done from `server.go`).  Given that this is the only property on this object it makes me wonder why `blockProcessor` exists as a type at all.

Also strange is that the block in `server.go` that does read from this, only uses stuff (`peer`) that is already on this object.

I suggest either removing `blockProcessor` as a type entirely (and just pass the list around), or move the below block from `server.go` into this type (in a `(bp *blockProcessor) ProcessBlocks()` func or similar):
```
for e := bp.composites.Front(); e != nil; e = e.Next() {
			nd := e.Value.(format.Node)
			err := s.peer.processBlock(ctx, txn, col, dsKey, nd, "")
			if err != nil {
				log.ErrorE(
					ctx,
					"Failed to process block",
					err,
					logging.NewKV("DocKey", dsKey.DocKey),
					logging.NewKV("CID", nd.Cid()),
				)
			}
		}
```

Reply:
I move the block to process.go and simplified things a little more. `blockProcessor` had more fiels initially but I was able make the change leaner and it got kept around. I added some relevant fields to help pass things around. I like having the `blockProcessor` type as it keeps the processing under a specific domaine.